src/coreclr/src/binder/inc/applicationcontext.hpp

```diff
@@ -46,7 +46,11 @@ namespace BINDER_SPACE
             key = e.m_wszSimpleName;
             return key;
         }
-        static count_t Hash(const key_t &str) { return HashiString(str); }
+        static count_t Hash(const key_t &str)
+        {
+            SString ssKey(str);
+            return ssKey.HashCaseInsensitive();
+        }
         static BOOL Equals(const key_t &lhs, const key_t &rhs) { LIMITED_METHOD_CONTRACT; return (_wcsicmp(lhs, rhs) == 0); }
```

Comment:
```suggestion
        static BOOL Equals(const key_t &lhs, const key_t &rhs) { LIMITED_METHOD_CONTRACT; return (SString::_wcsicmp(lhs, rhs) == 0); }
```
to make hash and equals consistent.